core/oa_hash_map.h

```diff
@@ -190,6 +191,40 @@ class OAHashMap {
 	_FORCE_INLINE_ uint32_t get_capacity() const { return capacity; }
 	_FORCE_INLINE_ uint32_t get_num_elements() const { return num_elements; }
 
+	bool empty() const {
+		return num_elements == 0;
+	}
+
+	void clear() {
+
+		for (Iterator it = iter(); it.valid; it = next_iter(it)) {
+			remove(*it.key);
+			it = next_iter(it);
+		}
```

Comment:
This loop looks fishy (you are stepping `it` twice); can probably be optimized as well.

Inlining the iterator loop, setting `DELETED_HASH_BIT` on all hashes and calling `~TValue` and `~TKey` directly should be a bit more obvious and a bit faster.

Reply:
Indeed the duplicated next_iter bit is from something I'd accidentally left over when refactoring.

I thought about the second bit but wasn't sure if it was worth doing it, I'll do that instead though now then :)